arch/riscv/boot/dts/bouffalolab/bl808.dtsi

```diff
@@ -140,5 +140,11 @@
 			#interrupt-cells = <2>;
 			riscv,ndev = <64>;
 		};
+
+                wdt: wdt@2000a500 {
+                        compatible = "bflb,bflb808-wdt";
+                        reg = <0x2000a500 0x100>;
```

Comment:
This MMIO range is for the TIMER0 device, not just for the watchdog portion, so it's not really correct to label the whole device with a "wdt" compatible string.

Reply:
This is my first device driver. Happy to make any change but I'm not sure what to change. I'm also not sure what happens if/when the other timers are implemented in a different driver module.

Reply:
The call to `devm_ioremap_resource` (or `devm_platform_ioremap_resource`) claims/reserves the MMIO range for use by the driver. You can see these claimed MMIO ranges in `/proc/iomem`. So if both a timer driver and a watchdog driver declare the same MMIO range, the second-loaded driver will fail to probe.

Supporting multiple subsystems with overlapping MMIO ranges requires cooperation between the drivers. There are multiple ways of doing this, and it's not obvious which is best in this situation:
1) Use a single driver for both subsystems
2) Use a MFD
3) Use a syscon
4) Use the auxiliary device bus

So I'm also not sure what to change here.

Reply:
Unfortunately the TCCR clock source register is shared between both general purpose timers and the WDT, so the drivers also need to consider coordinating sharing the same register, not just sets of register in the same MMIO space.